src/main/java/org/spongepowered/common/data/SpongeDataManager.java

```diff
@@ -385,4 +429,174 @@ public static void finalizeRegistration() {
         return Optional.ofNullable((ValueProcessor<E, ? extends BaseValue<E>>) this.valueDelegates.get(key));
     }
 
+    /**
+     * Registers a {@link DataManipulator} class and the
+     * {@link ImmutableDataManipulator} class along with the implemented
+     * classes such that the processor is meant to handle the implementations
+     * for those specific classes.
+     *
+     * @param manipulatorClass The manipulator class
+     * @param implClass The implemented manipulator class
+     * @param immutableDataManipulator The immutable class
+     * @param implImClass The implemented immutable class
+     * @param processor The processor
+     * @param <T> The type of data manipulator
+     * @param <I> The type of immutable data manipulator
+     */
+    public <T extends DataManipulator<T, I>, I extends ImmutableDataManipulator<I, T>> void registerContextDataProcessorAndImpl(
+        Class<T> manipulatorClass, Class<? extends T> implClass, Class<I> immutableDataManipulator,
+        Class<? extends I> implImClass, ContextDataProcessor<T, I> processor) {
+        checkState(allowRegistrations, "Registrations are no longer allowed!");
+        checkArgument(!Modifier.isAbstract(implClass.getModifiers()), "The Implemented DataManipulator class cannot be abstract!");
+        checkArgument(!Modifier.isInterface(implClass.getModifiers()), "The Implemented DataManipulator class cannot be an interface!");
+        checkArgument(!Modifier.isAbstract(implImClass.getModifiers()), "The implemented ImmutableDataManipulator class cannot be an interface!");
+        checkArgument(!Modifier.isInterface(implImClass.getModifiers()), "The implemented ImmutableDataManipulator class cannot be an interface!");
+        checkArgument(!(processor instanceof ContextDataProcessorDelegate), "Cannot register ContextDataProcessorDelegates!");
+
+        if (!this.contextInterfaceToImplDataManipulatorClasses.containsKey(manipulatorClass)) { // we only need to insert it once.
```

Comment:
Can simply check that the normal interface to implementation class is registered already.